xbrowse_server/api/views.py

```diff
@@ -661,6 +662,86 @@ def add_or_edit_variant_tags(request):
     })
 
 
+@login_required
+@log_request('delete_gene_note')
+def delete_gene_note(request, note_id):
+    ret = {
+        'is_error': False,
+    }
+
+    notes = GeneNote.objects.filter(id=note_id)
+    if not notes:
+        ret['is_error'] = True
+        ret['error'] = 'note id %s not found' % note_id
+    else:
+        note = list(notes)[0]
+        if not note.can_edit(request.user):
+            raise PermissionDenied
+
+        note.delete()
+
+    return JSONResponse(ret)
+
+@login_required
+@log_request('add_or_edit_gene_note')
+def add_or_edit_gene_note(request):
+    """Add a gene note"""
+
+    form = api_forms.GeneNoteForm(request.GET)
+    if not form.is_valid():
+        return JSONResponse({
+            'is_error': True,
+            'error': server_utils.form_error_string(form)
+        })
+
+    if form.cleaned_data.get('note_id'):
+        event_type = "edit_gene_note"
+
+        notes = GeneNote.objects.filter(
```

Comment:
would again use `GeneNote.objects.get` instead of `GeneNote.objects.filter`, and query just by note_id since that should be unique. Maybe it can go in a `get_note_or_error(..)` func. that does both the lookup and permissions check.